File: Backend-POS/POS.Main/POS.Main.Business.Admin/Models/Dashboard/SalesReportResponseModel.cs
namespace POS.Main.Business.Admin.Models.Dashboard;

public class SalesReportResponseModel
{
    public DashboardKpiModel Summary { get; set; } = new();
    public List<DailyBreakdownModel> DailyBreakdown { get; set; } = new();
    public List<CategoryBreakdownModel> CategoryBreakdown { get; set; } = new();
    public List<KitchenBreakdownModel> KitchenBreakdown { get; set; } = new();
}

public class DailyBreakdownModel
{
    public DateTime Date { get; set; }
    public decimal TotalSales { get; set; }
    public int OrderCount { get; set; }
    public int GuestCount { get; set; }
    public decimal AveragePerOrder { get; set; }
}

public class CategoryBreakdownModel
{
    public int CategoryType { get; set; }
    public string CategoryName { get; set; } = string.Empty;
    public decimal TotalSales { get; set; }
    public decimal Percentage { get; set; }
}

File: Frontend-POS/RBMS-POS-Client/src/app/core/api/models/sales-report-response-model.ts
/* tslint:disable */
/* eslint-disable */
import { CategoryBreakdownModel } from '../models/category-breakdown-model';
import { DailyBreakdownModel } from '../models/daily-breakdown-model';
import { DashboardKpiModel } from '../models/dashboard-kpi-model';
import { KitchenBreakdownModel } from '../models/kitchen-breakdown-model';
export interface SalesReportResponseModel {
  categoryBreakdown?: Array<CategoryBreakdownModel> | null;
  dailyBreakdown?: Array<DailyBreakdownModel> | null;
  kitchenBreakdown?: Array<KitchenBreakdownModel> | null;
  summary?: DashboardKpiModel;
}
